aten/src/ATen/native/Distance.cpp

```diff
@@ -42,12 +42,36 @@ Tensor cdist(const Tensor& x1, const Tensor& x2, const double p) {
 
   int64_t r1 = x1.size(-2);
   int64_t r2 = x2.size(-2);
-  Tensor result = at::empty({r1, r2}, x1.options());
+  auto dim1 = x1.dim();
+  auto dim2 = x2.dim();
+
+  IntArrayRef batch_tensor1(x1.sizes().data(), std::max<int64_t>(dim1 - 2, 0));
+  IntArrayRef batch_tensor2(x2.sizes().data(), std::max<int64_t>(dim2 - 2, 0));
+  std::vector<int64_t> expand_batch_portion = infer_size(batch_tensor1, batch_tensor2);
+  std::vector<int64_t> tensor1_expand_size(expand_batch_portion);
+  tensor1_expand_size.insert(tensor1_expand_size.end(), {r1, c1});
+  std::vector<int64_t> tensor2_expand_size(expand_batch_portion);
+  tensor2_expand_size.insert(tensor2_expand_size.end(), {r2, c2});
+
+  int expand_batch_product = std::accumulate(expand_batch_portion.begin(), expand_batch_portion.end(), 1, std::multiplies<int64_t>());
+  std::vector<int64_t> tensor1_view({expand_batch_product});
+  tensor1_view.insert(tensor1_view.end(), {r1, c1});
+  std::vector<int64_t> tensor2_view({expand_batch_product});
+  tensor2_view.insert(tensor2_view.end(), {r2, c2});
+
+  Tensor tensor1_expanded = x1.expand(tensor1_expand_size).contiguous().view(tensor1_view);
+  Tensor tensor2_expanded = x2.expand(tensor2_expand_size).contiguous().view(tensor2_view);
+
+  std::vector<int64_t> output_shape(expand_batch_portion);
```

Comment:
Why do sometimes you use `foo.insert(foo.end(), {bar, quux});` and sometimes you use `foo.push_back` twice? Should be consistent.

Reply:
Made it consistent